asv/environment.py

```diff
@@ -444,7 +444,9 @@ def install_project(self, conf, commit_hash=None):
         if commit_hash is None:
             commit_hash = self._cache.get_existing_commit_hash()
             if commit_hash is None:
-                commit_hash = self.repo.get_hash_from_master()
+                commit_hash = self.repo.get_hash_from_name(conf.branches[0])
```

Comment:
Probably needs to be `self.repo.get_hash_from_name(self.repo.get_branch_range_spec(conf.branches[0]))`